src/components/Modal/AddName/index.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useState } from "react";
+import {
+	ButtonsWrapper,
+	CancelButton,
+	CloseIconDiv,
+	Input,
+	InputWrap,
+	Label,
+	ModalTitle,
+	Wrapper,
+} from "./styles";
+import { ErrorMessage, StartButton } from "../../CategoryCard/styles";
+import { useQuizData } from "../../../context/quizContext";
+import { CLOSE__MODAL, SET__PLAYER__NAME } from "../../../constants";
+import { ReactComponent as CloseIcon } from "../../../icons/close-icon.svg";
+import { useHistory } from "react-router";
+
+function AddName() {
+	const { dispatch } = useQuizData();
+	const [name, setName] = useState("");
+	const [isNameActive, setIsNameActive] = useState(false);
```

Comment:
Just a suggestion here , since you are updating more than one state here (name & isNameActive) , I think this can be put inside an useReducer

Reply:
These two states are page-specific states so they can be used in a reducer.